server/http_handler_test.go

```diff
@@ -72,7 +72,6 @@ func createBasicHTTPHandlerTestSuite() *basicHTTPHandlerTestSuite {
 }
 
 func TestRegionIndexRange(t *testing.T) {
-	t.Parallel()
```

Comment:
I don't get the point why the `t.Parallel()` should be removed: according to https://github.com/pingcap/tidb/pull/29374, the case was included by `HTTPHandlerTestSuite` which is not a `SerialTestSuite`.

Reply:
Do you mean that I need to change only the cases which run serially before the testing, instead of all parallel tests in these 2 files?

Reply:
Here is the thing: `TestAllHistory` also fails(https://github.com/pingcap/tidb/issues/30256), but it ran in parallel all the time.

Reply:
> Do you mean that I need to change only the cases which run serially before the testing, instead of all parallel tests in these 2 files?

Theoretically, yes. However, it does not harm if we run all of them in serial, as long as the total time of it is acceptable, I want to confirm it with you firstly.

Reply:
I reverted those tests and let's take some trials.